x-pack/test/api_integration/apis/ml/results/get_anomaly_search.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const ml = getService('ml');
+  const spacesService = getService('spaces');
+  const supertest = getService('supertestWithoutAuth');
+
+  const adJobId = 'fq_single';
+  const idSpace1 = 'space1';
+  const idSpace2 = 'space2';
+  const defaultSpaceId = 'default';
+
+  const jobQuery = {
+    size: 1,
+    body: {
+      query: {
+        bool: {
+          filter: [{ term: { job_id: adJobId } }],
+        },
+      },
+    },
+  };
+
+  async function runRequest(
+    requestBody: {
+      jobIds: string[];
+      query: any;
+    },
+    space: string,
+    expectedStatusCode: number,
+    user: USER
+  ) {
+    const { body, status } = await supertest
+      .post(`/s/${space}/api/ml/results/anomaly_search`)
+      .auth(user, ml.securityCommon.getPasswordForUser(user))
+      .set(COMMON_REQUEST_HEADERS)
+      .send(requestBody);
+    ml.api.assertResponseStatusCode(expectedStatusCode, status, body);
+
+    return body;
+  }
+
+  describe('POST results/anomaly_search', () => {
+    before(async () => {
+      await ml.api.cleanMlIndices();
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/farequote');
+
+      // create spaces
+      await spacesService.create({ id: idSpace1, name: 'space_one', disabledFeatures: [] });
+      await spacesService.create({ id: idSpace2, name: 'space_two', disabledFeatures: [] });
+
+      await ml.api.createAndRunAnomalyDetectionLookbackJob(
+        ml.commonConfig.getADFqSingleMetricJobConfig(adJobId),
+        ml.commonConfig.getADFqDatafeedConfig(adJobId)
+      );
+      // reassign spaces for job
+      await ml.api.updateJobSpaces(adJobId, 'anomaly-detector', [idSpace1], []);
+      await ml.api.assertJobSpaces(adJobId, 'anomaly-detector', [defaultSpaceId, idSpace1]);
```

Comment:
you can pass space id as a third argument 
```suggestion
        ml.commonConfig.getADFqDatafeedConfig(adJobId),
        idSpace1
      );
```

Reply:
Updated in https://github.com/elastic/kibana/pull/142454/commits/d3e417aea9027f9b6612285293df19882f4b4fad